packages/symantec_endpoint_security/data_stream/event/elasticsearch/ingest_pipeline/pipeline_object_cybox.yml

```
@@ -156,27 +156,33 @@ processors:
           tag: append_cybox_files_accessed_into_file_accessed
           value: '{{{_ingest._value.accessed}}}'
           allow_duplicates: false
-  - foreach:
-      field: ses.cybox.files
+
+  - script:
+      description: Convert attribute ids to strings
+      tag: script_to_convert_attribute_ids
       if: ctx.ses?.cybox?.files instanceof List
-      tag: foreach_cybox_files
-      processor:
-        foreach:
-          field: _ingest._value.attribute_ids
-          tag: foreach_cybox_files_attribute_ids
-          processor:
-            convert:
-              field: _ingest._value
-              tag: convert_cybox_files_attribute_ids_to_string
-              type: string
-              ignore_missing: true
-              on_failure:
-                - remove:
-                    field: _ingest._value
-                    ignore_missing: true
-                - append:
-                    field: error.message
-                    value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+
+      lang: painless
+      source: >-
+          for (int i = 0; i < ctx.ses.cybox.files.length; i++) {
+            def file = ctx.ses.cybox.files[i];
+            if (file.containsKey('attribute_ids') && file.attribute_ids != null) {
+              def new_ids = [];
+              for (int j = 0; j < file.attribute_ids.length; j++) {
+                def id = file.attribute_ids[j];
+                if (id != null) {
+                  new_ids.add(id.toString());
+                }
+              }
+              file.attribute_ids = new_ids;
+            }
+          }
```

```suggestion
          for (int i = 0; i < ctx.ses.cybox.files.length; i++) {
            def file = ctx.ses.cybox.files[i];
            if (!file.containsKey('attribute_ids') || file.attribute_ids == null) {
              continue;
            }
            file.attribute_ids = [];
            for (int j = 0; j < file.attribute_ids.length; j++) {
              def id = file.attribute_ids[j];
              if (id != null) {
                file.attribute_ids.add(id.toString());
              }
            }
          }
```
- shift work left
- less intermediates (could keep the intermediate if it's only conditionally assigned to `file.attribute_ids`)

---

I see what you are trying to accomplish. I don't think the example will work.
file.attributes_ids is reassigned to an empty array, then you are trying to access items in the now empty array. 
`          file.attribute_ids = [];
            for (int j = 0; j < file.attribute_ids.length; j++)`
I would do a direct replacement in the array if I didn't need to check for null.